pkg/validation/sessions.go

```diff
@@ -38,6 +43,96 @@ func validateSessionCookieMinimal(o *options.Options) []string {
 	return msgs
 }
 
+func validateDynamoDBSessionStore(o *options.Options) []string {
+	if o.Session.Type != options.DynamoDBSessionType {
+		return []string{}
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	})
+	if err != nil {
+		return []string{fmt.Sprintf("unable to create dynamoDB client: %v", err)}
+	}
+
+	nonce, err := encryption.Nonce()
+	if err != nil {
+		return []string{fmt.Sprintf("unable to generate a dynamoDB initialization test key: %v", err)}
+	}
+
+	key := fmt.Sprintf("%s-healthcheck-%s", o.Cookie.Name, nonce)
+	return sendDynamoDBConnectionTest(dynamo.New(sess), o.Session.DynamoDB.TableName, key, nonce)
+}
+
+func sendDynamoDBConnectionTest(service *dynamo.DynamoDB, tableName string, key string, nonce string) []string {
```

Comment:
We should create a few helper methods for the main logic points inside of this. It is quite large all together and I think this would probably flag a few CodeClimate complexity alerts.

Reply:
Ah yep 😄 

https://codeclimate.com/github/oauth2-proxy/oauth2-proxy/pull/876